metabase/metabase
# Action Form Validation

resolves [#26579](https://github.com/metabase/metabase/issues/26579)

## Description
- Uses new `optional` prop in action form components from https://github.com/metabase/metabase/pull/27246
- auto-generates `Yup` form validation schema from form settings, and enforces form validation
- removes bespoke `setDefaultValues` and `setNumericValues` functions from form submission in favor of the `Yup` implementations

![Screen Shot 2022-12-15 at 2 46 41 PM](https://user-images.githubusercontent.com/30528226/207974204-2d6cfe0d-ca59-413a-88a7-fdf37ba0ea2d.png)

![liveupdate](https://user-images.githubusercontent.com/30528226/207974492-cd0a13d5-d870-4ba3-b40b-552c86c2b6fd.gif)


## Testing steps: implicit action forms
- Within a data app, add a scaffolded page
- click on one of the items to go to the detail page
- open up the edit form for that item
- the form should correctly identify which fields are optional (from the db)
- the form should not be submittable with any required fields empty
- the form should be submittable with required fields filled

## Testing steps: custom action forms
- Open a custom action with a query with one or more parameters
- see that toggling the required field setting in fact toggles the (optional) label int he preview
- add the action to a dashboard, and see that the optional labels appear appropriately
- see that the form can only be submitted when all optional fields are provided
- see that linking a dashboard filter to one or more fields in the form still allows the form to be submitted.
